### DISCUSSION TITLE
Build issues with NextJS14 `type` checking fails for `null` values

### DISCUSSION THREAD

**Discussion by devraj** [orval-labs/orval#1020] [Q&A]
I have an endpoint written in `python` using `fastapi` which exposes a standard OAuth token endpoint, which uses a OAuth2 Password Form model, expected to be passed in as form data.

```python
@router.post(
    "/token",
    summary="Provides an endpoint for login via email and password",
)
async def login_for_auth_token(
    form_data: OAuth2PasswordRequestForm = Depends(),
    session: AsyncSession = Depends(get_async_session)
) -> Token:
    """ Attempt to authenticate a user and issue JWT token

    If the user does not exists or the password is incorrect
    then an exception is raised which returns a 4xx response.

    """
    user = await User.get_by_email(
        session,
        form_data.username
    )
```
See [fastapi example](https://fastapi.tiangolo.com/tutorial/security/simple-oauth2/) for context.

I have used the same setup with frameworks like `remix` and not encountered any issues, however we are moving to NextJS14 and When I use orval to generate the client to consume the token endpoint `eslint` complains with the following:

```tsx
./api/auth/auth.ts:171:41
Type error: Argument of type 'BodyLoginForAuthTokenGrantType' is not assignable to parameter of type 'string'.
  Type 'null' is not assignable to type 'string'.

  169 |   const formUrlEncoded = new URLSearchParams();
  170 |   if (bodyLoginForAuthToken.grant_type !== undefined) {
> 171 |     formUrlEncoded.append("grant_type", bodyLoginForAuthToken.grant_type);
      |                                         ^
  172 |   }
  173 |   formUrlEncoded.append("username", bodyLoginForAuthToken.username);
  174 |   formUrlEncoded.append("password", bodyLoginForAuthToken.password);
```

adding a check for those values to be `!==null` makes `eslint` happy:

```tsx
  if (bodyLoginForAuthToken.grant_type !== undefined && bodyLoginForAuthToken.grant_type !== null) {
    formUrlEncoded.append("grant_type", bodyLoginForAuthToken.grant_type);
  }
```

I was wonder if this should be something `orval` generates? I can get around it by asking eslint to ignore the file, it would be nice to get someone else's thoughts on this.

Thank you for time.

---

Node version: v21.1.0
OS: macOS 14
Framework: NextJS 14.0.1
orval: 6.19.1

**Comment by melloware** [comment]
Interesting it sounds like a bug ?  Weird that it's saying can't assign Null to string but it's not NULL it's a String. Open to a fix if you want to investigate further.

**Reply by devraj** [reply]
As in lodge an issue or you'd like me to investigate the code and provide a pull request?

**Reply by melloware** [reply]
Investigate and PR

**Reply by devraj** [reply]
I intend to get to this in the next couple of days
